File: LeetCode2.java
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode ans = new ListNode();
        ListNode cur = ans;
        int add = 0;
        while(l1 != null || l2 != null){
            int sum = add + (l1 != null ? l1.val : 0) + (l2 != null ? l2.val : 0);
            add = sum/10;
            sum = sum%10;
            cur.next = new ListNode(sum);
            cur = cur.next;
            l1 = l1 == null ? null : l1.next;
            l2 = l2 == null ? null : l2.next;
        }
        if(add > 0)
            cur.next = new ListNode(add, null);
        return ans.next;
    }
}

File: python/LeetCode2.py
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        ans = ListNode()
        cur = ans
        prev = 0
        while l1 != None or l2 != None:
            cur.next = ListNode(((l1.val if l1 != None else 0) + (l2.val if l2 != None else 0) + prev) % 10)
            prev = ((l1.val if l1 != None else 0) + (l2.val if l2 != None else 0) + prev) // 10
            l1 = l1.next if l1 != None else None
            l2 = l2.next if l2 != None else None
            cur = cur.next

        if prev != 0:
            cur.next = ListNode(prev)

        return ans.next
